### PR TITLE
Allow query strings within the address/search bar.

### PR DESCRIPTION
**Pull request by the-broz**
This PR allows the searchbar to send query strings to remote servers (non-github urls because github discards them) and can be accessed with a new global lua function called "get_parameters" which returns a key-value table with all user parameters. Automatically redirects invalid urls to dingle.it for browsing.
This solves #19

### CODE REVIEW

File: napture/src/globals.rs [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
+use std::collections::HashMap;
 use lazy_static::lazy_static;
 
 lazy_static! {
     pub static ref LUA_LOGS: Mutex<String> = Mutex::new(String::new());
+}
+
+lazy_static! {
+    pub static ref URI_PARAMETERS: Mutex<HashMap<String, String>> = Mutex::new(HashMap::new());
 }
```
Comment (GStudiosX2) [inline napture/src/globals.rs:11]: Can't you have multiple in lazy_static?

File: napture/src/main.rs [hunk 2]
```diff
@@ -374,10 +411,14 @@ fn fetch_dns(url: String) -> String {
         let status = response.status();
 
         if let Ok(json) = response.json::<DomainInfo>() {
-            json.ip
+            return json.ip
```
Comment (GStudiosX2) [inline napture/src/main.rs:414]: Why did you add return here ?

File: napture/src/main.rs [hunk 3]
```diff
@@ -374,10 +411,14 @@ fn fetch_dns(url: String) -> String {
...
         } else {
             lualog!("debug", format!("Failed to parse response body from DNS API. Error code: {}. Returning original URL.", status.as_u16()));
-            url
+            if status.as_u16() == 404 {
+                URI_PARAMETERS.lock().unwrap().insert("q".to_string(), url.clone());
+                return "https://github.com/face-hh/dingle-frontend".to_string();
+            }
```
Comment (GStudiosX2) [inline napture/src/main.rs:420]: What about other http error statuses ?

**Review by GStudiosX2 — CHANGES_REQUESTED** [review]

File: napture/src/main.rs (hunk 3 above)
Comment (the-broz) [reply napture/src/main.rs:420]: The reason I didn't handle any other http statuses is because if it exists and something is wrong with the server we don't want to act like it doesn't exist.

File: napture/src/main.rs (hunk 2 above)
Comment (the-broz) [reply napture/src/main.rs:414]: Did that out of habit of other languages lol

File: napture/src/globals.rs (hunk 1 above)
Comment (the-broz) [reply napture/src/globals.rs:11]: Didn't know that

File: napture/src/main.rs (hunk 3 above)
Comment (GStudiosX2) [reply napture/src/main.rs:420]: Ah okay but probably should bring up some sort of error page

Comment (the-broz) [reply napture/src/main.rs:420]: Good point.

**Comment by the-broz** [comment]
accidentally broke the entire searchbar my bad

**Comment by the-broz** [comment]
This was added in a different way in the main version closing.
